src/Modules/CalcOffence.lua

```diff
@@ -4125,20 +4136,26 @@ function calcs.offence(env, actor, activeSkill)
 			local igniteStacks = 1
 			if not skillData.triggeredOnDeath then
 				if output.Cooldown then
-					igniteStacks = (globalOutput.IgniteDuration / m_max(output.Cooldown, (output.HitTime or output.Time)) * skillData.dpsMultiplier) / maxStacks
+					igniteStacks = ((output.HitChance / 100) * globalOutput.IgniteDuration / m_max(output.Cooldown, (output.HitTime or output.Time)) * skillData.dpsMultiplier) / maxStacks
```

Comment:
I'm pretty sure dividing by maxStacks here and than calculating 

`local igniteRollAverage = igniteStacks / (igniteStacks + 1) * 100`

is not correct.

If we apply 10 stacks per duration and have maximum stacks of 2, current calculations give 
`stack potential = 10 / 2 == 5`
`igniteRollAverage =  5/6 * 100 == 83.3%`

while the correct value is the average of 10/11 and 9/10 == 90.45% -- two most powerful ignites currently.

It also undervalues ignites with low uptime. If your ignite duration is 0.1s and you can apply 1 stack per second igniteStacks will be 0.1 resulting in average roll of 9% which is not true, average roll will be 50%

Reply:
Also was looking at https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues/6272 and realized that igniteStacks here should also address ignite chance. Otherwise reducing ignite chance increases damage as potential stays unchanged and calcs think that more of ignites are from crits now

Reply:
I agree with your comments and will fix them for a future PR but I need to merge this now for the next update